### PR TITLE
Course type merge UI

### PR DESCRIPTION
**Pull request by janno42**
allow merging course types via the UI

### CODE REVIEW

File: evap/staff/templates/staff_course_type_index.html [hunk 1]
```diff
@@ -52,9 +52,8 @@
             </tbody>
         </table>
         <div class="form-group well">
-            <div class="col-sm-6">
-                <input type="submit" value="{% trans "Save course types" %}" class="btn btn-primary"/>
-            </div>
+            <input type="submit" value="{% trans "Save course types" %}" class="btn btn-primary"/>
+            <a href="{% url "staff:course_type_merge_selection" %}" class="btn btn-default">{% trans "Merge course types" %}</a>
```
Comment (karyon) [inline evap/staff/templates/staff_course_type_index.html]: isn't that a bit weird, a button below a form that actually has nothing to do with the form?

File: evap/staff/templates/staff_course_type_merge.html [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    {{ block.super }}
+    <li><a href="{% url "staff:course_type_index" %}">{% trans "Course types" %}</a></li>
+    <li>{% trans "Merge course types" %}</li>
+{% endblock %}
+
+{% block content %}
+    {{ block.super }}
+    <h2>{% trans "Merge course types" %}</h2>
+
+    <p>{% blocktrans %}The following courses currently have the course type <strong>{{ other_type }}</strong> and will get the type <strong>{{ main_type }}</strong> once you merge the course types. The course type <strong>{{ other_type }}</strong> will be deleted after that.{% endblocktrans %}</p>
```
Comment (karyon) [inline evap/staff/templates/staff_course_type_merge.html]: line break please

File: evap/staff/templates/staff_course_type_index.html (hunk 1 above)
Comment (janno42) [inline evap/staff/templates/staff_course_type_index.html]: would you prefer it above the list of course types?

File: evap/staff/views.py [hunk 3]
```diff
@@ -855,6 +855,37 @@ def course_type_index(request):
...
+
+@staff_required
+def course_type_merge(request, main_type_id, other_type_id):
+    main_type = get_object_or_404(CourseType, id=main_type_id)
+    other_type = get_object_or_404(CourseType, id=other_type_id)
+
+    if request.method == 'POST':
+        for course in Course.objects.filter(type=other_type):
+            course.type = main_type
+            course.save()
```
Comment (karyon) [inline evap/staff/views.py]: please use the update method here

File: evap/staff/templates/staff_course_type_index.html (hunk 1 above)
Comment (karyon) [inline evap/staff/templates/staff_course_type_index.html]: not 100% sure but i think i do.

File: evap/staff/views.py [hunk 4]
```diff
@@ -855,6 +855,37 @@ def course_type_index(request):
 
 
 @staff_required
+def course_type_merge_selection(request):
+    form = CourseTypeMergeSelectionForm(request.POST or None, request.FILES or None)
```
Comment (janno42) [inline evap/staff/views.py]: copy paste. removed.

File: evap/staff/templates/staff_course_type_merge.html (hunk 2 above)
Comment (janno42) [inline evap/staff/templates/staff_course_type_merge.html]: check

File: evap/staff/views.py (hunk 3 above)
Comment (janno42) [inline evap/staff/views.py]: check

**Comment by janno42** [comment]
updated

**Comment by karyon** [comment]
:house_with_garden:

**Pull request merged**